Yoast/wordpress-seo
# Fix highlighting words functionality in Japanese

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adapts the functionality to mark words in a sentence for Japanese.

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

### **In content-analysis app** (to test different word forms)

* Enable JAPANESE_SUPPORT in `specHelpers/getMorphologyData.js` and in `packages/yoastseo/src/languageProcessing/helpers/sentence/getSentences.js` (i.e. `enableFeatures( [ "JAPANESE_SUPPORT" ] );` (don't commit anything)
* Start the app (run `yarn start` in `wordpress-seo/apps/content-analysis`)
* Set the locale to `ja`
* Turn the `Use morphology` toggle on
* Turn on the `Use keyphrase distribution` toggle on
* Create a post with at least 15 sentences (use [this generator](https://generator.lorem-ipsum.info/_japanese) to generate Japanese text)
* Set `猫の餌` as the keyphrase
* Add this sentence `どちらもとても可愛くて甘い猫で、猫の餌を食べるのが大好きです。` to the text
* Confirm that the keyphrase is recognized in the Keyphrase distribution assessment and that the recognized keyphrase is highlighted
* Add these words as the synonyms `猫用のフード , 猫用食品`
* Add this sentences to the text: `彼らが好きなタイプの猫用フードは新鮮なものです。加工が少ない猫用食品の一種。`
* Confirm that the synonyms is recognized in the Keyphrase distribution assessment and that the recognized synonyms are also highlighted
* Change your keyphrase to `小さい花の刺繍` and remove the synonyms from the synonyms field
* Add this sentence to your text: `小さくて可愛い花の刺繍に関する一般一般の記事です。`
* Confirm that `小さく`, `花` and `刺繍` are highlighted when the eye icon of the keyphrase distribution assessment is clicked
* Add this sentence to your text: `この記事は小さい花の刺繍をどうやってすてればいいのか、基本的な情報を紹介します。`
* Confirm that `小さい`, `花` and `刺繍` are highlighted when the eye icon of the keyphrase distribution assessment is clicked

### **In Docker** (to test the feature flag)

**Requirements:**
* Build Free plugin
* Run `composer require yoast/wordpress-seo:dev-LINGO-1138-fix-highlighting-functionality-in-Japanese@dev` in `wordpress-seo-premium`
* Build Premium plugin

**Test with JAPANESE_SUPPORT enabled**

* Enable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in `basic-wordpress-config.php` file
* Set the site language to Japanese
* Create a post with at least 15 sentences (use [this generator](https://generator.lorem-ipsum.info/_japanese) to generate Japanese text)
* Set `猫の餌` as the keyphrase
* Add this sentence `どちらもとても可愛くて甘い猫で、猫の餌を食べるのが大好きです。` to the text
* Confirm that the keyphrase is recognized in the Keyphrase distribution assessment (or `キーフレーズ分布`) and that the recognized keyphrase is highlighted when the eye icon is clicked
* Add these words as the synonyms `猫用のフード , 猫用食品`
* Add this sentences to the text: `彼らが好きなタイプの猫用フードは新鮮なものです。加工が少ない猫用食品の一種。`
* Confirm that the synonyms are recognized in the Keyphrase distribution assessment (or `キーフレーズ分布`) and that the recognized synonyms are highlighted when the eye icon is clicked

**Test with JAPANESE_SUPPORT disabled**

* Disable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in `basic-wordpress-config.php` file
* Save your changes in the previous post and update it
* Confirm that the Keyphrase distribution assessment gives a grey bullet

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

**Requirements:**
* Install and activate Free and Premium plugin

**Test with JAPANESE_SUPPORT enabled**

* Enable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in your `wp-config.php` file for your setup
* Set the site language to Japanese
* Create a post with at least 15 sentences (use [this generator](https://generator.lorem-ipsum.info/_japanese) to generate Japanese text)
* Set `猫の餌` as the keyphrase
* Add this sentence `どちらもとても可愛くて甘い猫で、猫の餌を食べるのが大好きです。` to the text
* Confirm that the keyphrase is recognized in the Keyphrase distribution assessment (or `キーフレーズ分布`) and that the recognized keyphrase is highlighted when the eye icon is clicked
* Add these words as the synonyms `猫用のフード , 猫用食品`
* Add this sentences to the text: `彼らが好きなタイプの猫用フードは新鮮なものです。加工が少ない猫用食品の一種。`
* Confirm that the synonyms are recognized in the Keyphrase distribution assessment (or `キーフレーズ分布`) and that the recognized synonyms are highlighted when the eye icon is clicked

**Test with JAPANESE_SUPPORT disabled**

* Disable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in your `wp-config.php` file for your setup
* Save your changes in the previous post and update it
* Confirm that the Keyphrase distribution assessment gives a grey bullet

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-1138
